src/main/java/org/mockito/internal/matchers/CapturingMatcher.java

```diff
@@ -26,19 +34,33 @@ public String toString() {
     }
 
     public T getLastValue() {
-        if (arguments.isEmpty()) {
-            throw noArgumentValueWasCaptured();
-        }
-
-        return (T) arguments.getLast();
+        readLock.lock();
+        try {
+            if (arguments.isEmpty()) {
+                throw noArgumentValueWasCaptured();
+            }
 
+            return (T) arguments.get(arguments.size() - 1);
+        } finally {
+            readLock.unlock();
+        }
     }
 
     public List<T> getAllValues() {
-        return (List) arguments;
+        readLock.lock();
+        try {
+            return Arrays.asList((T[]) arguments.toArray());
```

Comment:
@szczepiq Hi! Thanks for reviewing these changes. I'll change this code to new ArrayList instantiation.